frontend/public/components/overview.jsx

```diff
@@ -0,0 +1,404 @@
+      || this.props.replicationControllers !== prevProps.replicationControllers
+      || this.props.pods !== prevProps.pods
+      || this.props.deploymentConfigs !== prevProps.deploymentConfigs
+      || this.props.routes !== prevProps.routes
+      || this.props.services !== prevProps.services) {
+      this.createOverviewData();
+    } else if (this.state.filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems,
+        groupedItems: this.groupByLabel(filteredItems, this.state.selectedGroupLabel)
+      });
+    } else if (this.state.selectedGroupLabel !== prevState.selectedGroupLabel) {
+      this.setState({
+        groupedItems: this.groupByLabel(this.state.filteredItems, this.state.selectedGroupLabel)
+      });
+    }
+  }
+
+  sortByDeploymentVersion(replicationControllers, descending) {
+    const compareDeployments = (left, right) => {
+      const leftVersion = parseInt(_.get(left, 'openshift.io/deployment-config.latest-version'), 10);
+      const rightVersion = parseInt(_.get(right, 'openshift.io/deployment-config.latest-version'), 10);
+      // Fall back to sorting by name if no deployment versions.
+      let leftName, rightName;
+      if (!_.isFinite(leftVersion) && !_.isFinite(rightVersion)) {
+        leftName = _.get(left, 'metadata.name', '');
+        rightName = _.get(right, 'metadata.name', '');
+        if (descending) {
+          return rightName.localeCompare(leftName);
+        }
+        return leftName.localeCompare(rightName);
+      }
+
+      if (!leftVersion) {
+        return descending ? 1 : -1;
+      }
+
+      if (!rightVersion) {
+        return descending ? -1 : 1;
+      }
+
+      if (descending) {
+        return rightVersion - leftVersion;
+      }
+
+      return leftVersion - rightVersion;
+    };
+
+    return _.toArray(replicationControllers).sort(compareDeployments);
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      const name = _.get(item, 'metadata.name', '');
+      return name.toLowerCase().includes(filterString);
+    });
+  }
+
+  groupByLabel(items, label) {
+    return _.chain(items)
+      .groupBy((item) => _.get(item, ['metadata', 'labels', label], 'other'))
+      .map((group, labelValue) => {
+        return {
+          name: labelValue,
+          items: group
+        };
+      })
+      .value();
+  }
+
+  getLabelGroupOptions(items) {
+    const {groupOptions} = this.state;
+    return _.chain(items)
+      .map(item => _.keys(item.metadata.labels))
+      .flatten()
+      .reduce((accumulator, key) => {
+
+        if (_.has(key, accumulator)) {
+          return accumulator;
+        }
+
+        return {
+          ...accumulator,
+          [key]: key
+        };
+      }, groupOptions)
+      .value();
```

Comment:
Wondering if this is easier if we use `Set`

Reply:
This is used in Dropdown. If dropdown accepts a Set or we refactor it to accept a Set, that would work.